src/generic/grid.cpp

```diff
@@ -11337,4 +11400,640 @@ wxGetContentRect(wxSize contentSize,
     return contentRect;
 }
 
+
+#if wxUSE_ACCESSIBILITY
+
+//-----------------------------------------------------------------------------
+// helpers for wxGridAccessible and wxGridCellAccessible
+//-----------------------------------------------------------------------------
+
+// helper to convert row / col into child object id
+// error handling is not required here
+static int GetChildId(wxGrid* grid, int row, int col,
+                      bool isRowHeader, bool isColHeader)
+{
+    int childId = 0;
+    int numberCols = grid->GetNumberCols();
+    if ( grid->GetRowLabelSize() )
+        numberCols++;
+
+    if ( isColHeader && isRowHeader )
+        childId = 1;
+    else if ( isColHeader )
+    {
+        childId = col + 1;
+        if ( grid->GetRowLabelSize() )
+            childId++;
+    }
+    else if ( isRowHeader )
+    {
+        childId = row * numberCols + 1;
+        if ( grid->GetColLabelSize() )
+            childId += numberCols;
+    }
+    else
+    {
+        childId = row * numberCols + col + 1;
+        if ( grid->GetColLabelSize() )
+            childId += numberCols;
+        if ( grid->GetRowLabelSize() )
+            childId++;
+    }
+    return childId;
+}
+
+// helper to calculate row, col, isRowHeader, isColHeader from childId
+static void GetRowCol(wxGrid* grid, int childId, int* row, int* col,
+                      bool* isRowHeader, bool* isColHeader)
```

Comment:
So I'd also like to get rid of `isXXXHeader` output parameters here and just set `row` and/or `col` to `-1`.

I also think that it could be nicer to return `wxGridCellCoords` from this function instead of using output parameters.